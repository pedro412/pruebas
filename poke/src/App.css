.App {
  max-width: 768px;
  margin: auto;
}