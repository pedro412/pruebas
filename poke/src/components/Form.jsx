import { useState } from 'react';

const apiURL = 'http://localhost:4000/api/v1/pokemon';

async function fetchPokemon(name) {
  try {
    const result = await fetch(`${apiURL}/${name}`);
    const data = await result.json();
    return data.pokemon;
  } catch (error) {
    return null;
  }
}

const Form = ({ setPokemons }) => {
  const [value, setValue] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();
    const pokemon = await fetchPokemon(value);
    if (pokemon) {
      pokemon.count = 1;
      setPokemons(pokemon);
    }
  };

  return (
    <>
      <h1>Form</h1>
      <form onSubmit={handleSubmit} id="form">
        <input
          name="name"
          onChange={(e) => setValue(e.target.value)}
          value={value}
          type="text"
        />
        <button>Catch</button>
      </form>
    </>
  );
};

export default Form;
