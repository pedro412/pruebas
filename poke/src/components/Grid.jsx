const Grid = ({ pokemons }) => {
  return (
    <>
      <h1>Grid</h1>

      {pokemons.length < 1 && (
        <div className="alert">No Pokemon captured yet</div>
      )}

      <div className="grid">
        {pokemons.map(({ name, id, image, count }) => (
          <div key={id} className="card">
            <h2>{name}</h2>
            <p>ID: {id}</p>
            <p>Count: {count}</p>
            <img src={image} alt={name} />
          </div>
        ))}
      </div>
    </>
  );
};

export default Grid;
