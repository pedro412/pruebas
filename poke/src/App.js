import { useState } from 'react';
import Form from './components/Form';
import Grid from './components/Grid';
import './App.css';

function App() {
  const [pokemons, setPokemons] = useState([]);

  const handleAddPokemon = (pokemon) => {
    const findPokemon = pokemons.find((e) => e.name === pokemon.name);
    if (findPokemon) {
      const pokemonsCopy = [...pokemons];
      const count = findPokemon.count;
      pokemonsCopy.find((e) => e.name === pokemon.name).count = count + 1;
      setPokemons(pokemonsCopy);
    } else {
      setPokemons([...pokemons, pokemon]);
    }
  };

  return (
    <div className="App">
      <Form setPokemons={(e) => handleAddPokemon(e)} />
      <Grid pokemons={pokemons} />
    </div>
  );
}

export default App;
