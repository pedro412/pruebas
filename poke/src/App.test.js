import App from './App';
import Form from './components/Form';
import Grid from './components/Grid';
const enzyme = require('enzyme');

describe('App', () => {
  const wrapper = enzyme.shallow(<App />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <Form /> tree', () => {
    expect(wrapper.find(Form)).toHaveLength(1);
  });

  it('should render <Grid /> tree', () => {
    expect(wrapper.find(Grid)).toHaveLength(1);
  });
});
