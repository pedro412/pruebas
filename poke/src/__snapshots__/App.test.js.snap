// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <Form
    setPokemons={[Function]}
  />
  <Grid
    pokemons={Array []}
  />
</div>
`;
